### PR TITLE
Yadda Secret Management

### PR DESCRIPTION
**Pull request by Sieabah**
This is pending [yadda-secret](https://github.com/Asymmetrik/yadda-secret/pull/1) being approved and published.

### CODE REVIEW

File: cli/commands/create-secret-senter.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+'use strict';
```
Comment (andrewmarcus) [inline cli/commands/create-secret-senter.js:1]: Senter with an S?

File: lib/secret-tasks.js [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+ * @return {module.exports|Credstash}
+ */
+function getCredentialStore(options){
+	var tableDetails = getCredentialTableDetails(options);
+	var store = new Credstash({
+		table: tableDetails.name,
+		awsOpts: { region: tableDetails.region },
+		kmsKey: options.DeploymentCenter.secret.kmsKeyAlias,
+	});
+	store.__name = tableDetails.name;
```
Comment (andrewmarcus) [inline lib/secret-tasks.js:30]: Why is this line necessary?

File: lib/secret-tasks.js [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+ * @return options
+ */
+exports.setupSecretCenter = function(options){
+	return Q.when(options)
+		.then(function(){
+			var app = options.AppName;
+			var env = options.environment;
+			var region = options.aws.region;
+
+			options.DeploymentCenter.secret._prefixKey = function(name) {
```
Comment (andrewmarcus) [inline lib/secret-tasks.js]: Would it make sense to surround this with null checks, or is it guaranteed to be defined by this point?

Comment (andrewmarcus) [reply lib/secret-tasks.js]: Looks like it is only guaranteed to be defined if `verify`  is called first, which only happens in the `create-secret-center` and `secret` commands, not in `deploy` or the other non-secret-specific commands.  Yadda needs to continue to work if secrets aren't defined.

File: lib/secret-tasks.js [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+
+/**
+ * Creates SecretCenter table in DynamoDB
+ * @param {object} options - user provided options
+ */
+exports.createSecretCenter = function(options){
+	return Q.when(options)
+		.then(function(){
+			logger.info('Creating DeploymentCenter Secret Center');
+			var store = getCredentialStore(options);
```
Comment (andrewmarcus) [inline lib/secret-tasks.js]: Why wouldn't you want to just use `options.DeploymentCenter.secret.secretStore` here, since `setupSecretCenter` has already been called?  Is there an advantage to creating the store again?

**Review by andrewmarcus — CHANGES_REQUESTED** [review]

File: lib/secret-tasks.js (hunk 2 above)
Comment (Sieabah) [reply lib/secret-tasks.js:30]: Just so I can retrieve the name of the table later. It's a double underscore name so it shouldn't be used, if ever.

File: lib/secret-tasks.js (hunk 4 above)
Comment (Sieabah) [reply lib/secret-tasks.js]: When I wrote this part I hadn't done the setupSecretCenter part yet. Good catch

File: lib/secret-tasks.js (hunk 3 above)
Comment (Sieabah) [reply lib/secret-tasks.js]: Good catch again.

File: cli/commands/create-secret-senter.js (hunk 1 above)
Comment (Sieabah) [reply cli/commands/create-secret-senter.js:1]: It may have been an easter egg

File: lib/secret-tasks.js [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+
+/**
+ * Creates SecretCenter table in DynamoDB
+ * @param {object} options - user provided options
+ */
+exports.createSecretCenter = function(options){
+	return Q.when(options)
+		.then(function(){
+			logger.info('Creating DeploymentCenter Secret Center');
+			var store = options.DeploymentCenter.secret.secretStore;
```
Comment (andrewmarcus) [inline lib/secret-tasks.js]: Isn't this a function?

File: lib/secret-tasks.js [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+ * @param {string} options.environment - Application environment
+ * @param {string} options.aws.region - Application region
+ * @param {object} options.DeploymentCenter.secret - Secret center configuration
+ * @return options
+ */
+exports.setupSecretCenter = function(options){
+	return Q.when(options)
+		.then(function(){
+			//Determine if secrets are enabled
+			if(options.DeploymentCenter.secret)
```
Comment (andrewmarcus) [inline lib/secret-tasks.js]: boolean logic fail?

**Review by andrewmarcus — CHANGES_REQUESTED** [review]

File: lib/secret-tasks.js (hunk 5 above)
Comment (Sieabah) [reply lib/secret-tasks.js]: Yes it is

File: lib/secret-tasks.js (hunk 6 above)
Comment (Sieabah) [reply lib/secret-tasks.js]: Should be good now

**Review by andrewmarcus — APPROVED** [review]
